src/main/java/org/jabref/gui/preferences/PreviewTabView.java

```diff
@@ -0,0 +1,328 @@
+        contextMenu.getItems().get(0).disableProperty().bind(viewModel.selectedIsEditableProperty().not());
+        contextMenu.getItems().get(2).disableProperty().bind(viewModel.selectedIsEditableProperty().not());
+
+        // editArea.editableProperty().bind(true); // FixMe: Cursor caret disappears
+
+        update();
+    }
+
+    private void update() {
+        PreviewLayout selectedLayout = viewModel.chosenSelectionModelProperty().getValue().getSelectedItem();
+        if (selectedLayout != null) {
+            String previewText;
+            try {
+                ((PreviewViewer) previewPane.getContent()).setLayout(selectedLayout);
+            } catch (StringIndexOutOfBoundsException exception) {
+                LOGGER.warn("Parsing error.", exception);
+                dialogService.showErrorDialogAndWait(Localization.lang("Parsing error"), Localization.lang("Parsing error") + ": " + Localization.lang("illegal backslash expression"), exception);
+            }
+
+            if (selectedLayout instanceof TextBasedPreviewLayout) {
+                previewText = ((TextBasedPreviewLayout) selectedLayout).getText().replace("__NEWLINE__", "\n");
+            } else {
+                previewText = ((CitationStylePreviewLayout) selectedLayout).getSource();
+            }
+            editArea.replaceText(previewText);
+        }
+    }
+
+    public void jumpToSearchKey(ListView<PreviewLayout> list, KeyEvent keypressed) {
+        if (keypressed.getCharacter() == null) {
+            return;
+        }
+
+        if (System.currentTimeMillis() - lastKeyPressTime < 1000) {
+            listSearchTerm += keypressed.getCharacter().toLowerCase();
+        } else {
+            listSearchTerm = keypressed.getCharacter().toLowerCase();
+        }
+
+        lastKeyPressTime = System.currentTimeMillis();
+
+        list.getItems().stream().filter(item -> item.getName().toLowerCase().startsWith(listSearchTerm))
+                .findFirst().ifPresent(list::scrollTo);
+    }
+
+    public void dragOverAvailable(@SuppressWarnings("unused") PreviewLayout layout, DragEvent event) {
+        if ((event.getGestureSource() != layout) && event.getDragboard().hasContent(DragAndDropDataFormats.PREVIEWLAYOUT)) {
+            event.acceptTransferModes(TransferMode.MOVE);
+        }
+    }
+
+    public void dragOverChosen(DragEvent event) {
+        if (event.getDragboard().hasContent(DragAndDropDataFormats.PREVIEWLAYOUT)) {
+            event.acceptTransferModes(TransferMode.MOVE);
+        }
+    }
+
+    public void dragDetected(ListView<PreviewLayout> sourceListView, MouseEvent event) {
+        PreviewLayout layout = sourceListView.getSelectionModel().getSelectedItem();
+        if (layout != null) {
+            ClipboardContent content = new ClipboardContent();
+            Dragboard dragboard = sourceListView.startDragAndDrop(TransferMode.MOVE);
+            content.put(DragAndDropDataFormats.PREVIEWLAYOUT, layout.getName());
```

Comment:
The PreviewLayout class mus implement the serializable interface.
Note that things like FXproperties or other observables can't be serialized, they must be marked as transient. 
See `LinkedFile` for an example.

Reply:
Ok, i tried making the classes and subclasses of CitationStylePreviewLayout and TextBasedPreviewLayout Serializable, in the Abbreviation-Class i had to make the StringProperties transient. Somehow java is still complaining about a non-serializable-object, but I can see none.

java.lang.IllegalArgumentException: Could not serialize the data
	at com.sun.javafx.tk.quantum.QuantumClipboard.putContentToPeer(QuantumClipboard.java:625) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumClipboard.flush(QuantumClipboard.java:269) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.startDrag(QuantumToolkit.java:1274) ~[jfxrt.jar:?]
	at javafx.scene.Scene$DnDGesture.dragDetectedProcessed(Scene.java:2953) ~[jfxrt.jar:?]
	at javafx.scene.Scene$DnDGesture.process(Scene.java:3022) ~[jfxrt.jar:?]
	at javafx.scene.Scene$DnDGesture.access$8200(Scene.java:2909) ~[jfxrt.jar:?]
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3773) ~[jfxrt.jar:?]
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485) ~[jfxrt.jar:?]
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762) ~[jfxrt.jar:?]
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295) ~[jfxrt.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_212]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$358(GlassViewEventHandler.java:432) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:410) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431) ~[jfxrt.jar:?]
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555) ~[jfxrt.jar:?]
	at com.sun.glass.ui.View.notifyMouse(View.java:937) ~[jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method) ~[jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication.lambda$null$152(WinApplication.java:177) ~[jfxrt.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
Caused by: java.io.NotSerializableException: sun.nio.cs.UTF_8
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348) ~[?:1.8.0_212]
	at java.util.ArrayList.writeObject(ArrayList.java:766) ~[?:1.8.0_212]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_212]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_212]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_212]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_212]
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1140) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178) ~[?:1.8.0_212]
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348) ~[?:1.8.0_212]
	at com.sun.javafx.tk.quantum.QuantumClipboard.putContentToPeer(QuantumClipboard.java:621) ~[jfxrt.jar:?]
	... 20 more

Reply:
Best is to debug and look at which property it fails. It's a bit trial and error, remember this from linked fiel

Reply:
Since the drag and drop is only JabRef-wide (i.e you don't need to support droping on different applications), it is easier to use the `CustomLocalDragboard` (which does not need to serialize the objects). Have a look here: https://github.com/JabRef/jabref/search?q=BIBENTRY_LIST_CLASS&unscoped_q=BIBENTRY_LIST_CLASS